src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -814,10 +835,29 @@ Node TheoryStringsRewriter::rewriteConcatRegExp(TNode node)
   retNode = utils::mkConcat(REGEXP_CONCAT, cvec);
   if (retNode != node)
   {
-    // handles all cases where consecutive re constants are combined, and cases
-    // where arguments are swapped, as described in the loop above.
+    // handles all cases where consecutive re constants are combined or dropped
+    // as described in the loop above.
     return returnRewrite(node, retNode, "re.concat");
   }
+
+  // flipping adjacent star arguments
+  changed = false;
+  for (unsigned i = 0, size = cvec.size(); (i + 1) < size; i++)
```

Comment:
```suggestion
  for (size_t i = 0, size = cvec.size() - 1; i < size; i++)
```
I find the change to `cvec.size() - 1` easier to read (just a suggestion though). Also, `size_t` is more appropriate here.